File: python_backend/ranges/jumps/filter_init_jumps.py
from sqr_case.get_sqr_case import get_sqr_case
from sqr_case.sqr_cases import *


def filter_init_jumps(jumps, board, color):
    """ """
    return list(filter(lambda j: get_sqr_case(board, j[0], j[1], color) != OOB
                          and get_sqr_case(board, j[0], j[1], color) != FRIEND,
                jumps))

File: node.js/ranges/jumps/filterInitJumps.js
var  getSqrCase  = require("../../sqrCase/getSqrCase");
const  {OOB, EMPTY, FRIEND, ENEMY}  = require("../../sqrCase/sqrCases");


function filterInitJumps(jumps, board, color) {
    /* **/
    return jumps.filter(j => getSqrCase(board, j[0], j[1], color) != OOB &&  getSqrCase(board, j[0], j[1], color) != FRIEND)
}

module.exports = filterInitJumps;